.github/workflows/duckdb-ref-update.yml

```diff
@@ -0,0 +1,53 @@
+name: duckdb-ref-update
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 4 * * 6'  # Every Saturday at 4 AM US time (America/New_York)
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.run_started_at }}"
+  cancel-in-progress: true
+
+jobs:
+  update-duckdb-refs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Fetch and Update All Refs
+        id: fetch-update-refs
+        shell: bash
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
```

Comment:
I think it would be good to skip the update if the duckdb ref hasn't updated. I don't think we want to update the extensions if we aren't updating dudckdb